azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/participant/grid/cell/ParticipantGridCellVideoView.kt

```diff
@@ -102,14 +104,21 @@ internal class ParticipantGridCellVideoView(
     }
 
     private fun setDisplayName(displayName: String) {
-        displayNameOnVideoTextView.text = displayName
+        if (displayName.isBlank()) {
+            displayNameOnVideoTextView.visibility = GONE
+        } else {
+            displayNameOnVideoTextView.text = displayName
+        }
     }
 
     private fun setMicButtonVisibility(isMicButtonVisible: Boolean) {
-
         if (!isMicButtonVisible) {
+            if (displayNameOnVideoTextView.visibility == GONE) {
```

Comment:
Is the visibility of another element the best way to express this condition?

Reply:
At the moment, yes, because the only way to "know" if the name is empty or not is to check if the name text view's visibility since we don't have access to displayName here. This is dependant on setDisplayName func where we set the text view's visibility to gone if the name is empty.

Reply:
Muhammad, instead of having logic in view, can we create view model properties that will notify for visibility? We can also write test cases.

Reply:
@iaulakh Since that will require creating view model, I will just merge this PR for the bug fix and send another PR for that.